programs/compressed-pda/src/event.rs

```diff
@@ -5,15 +5,23 @@ use anchor_lang::{
     solana_program::{instruction::Instruction, program::invoke},
 };
 
-use crate::{utxo::Utxo, InstructionDataTransfer, TransferInstruction};
+use crate::{
+    compressed_account::{CompressedAccount, CompressedAccountWithMerkleContext},
+    InstructionDataTransfer, TransferInstruction,
+};
 
 #[derive(Debug, Clone, AnchorSerialize, AnchorDeserialize)]
 pub struct PublicTransactionEvent {
-    pub in_utxos: Vec<Utxo>,
-    pub out_utxos: Vec<Utxo>,
-    pub de_compress_amount: Option<u64>,
-    pub out_utxo_indices: Vec<u64>,
+    pub input_compressed_account_hashes: Vec<[u8; 32]>,
+    pub output_account_hashes: Vec<[u8; 32]>,
+    pub input_compressed_accounts: Vec<CompressedAccountWithMerkleContext>,
+    pub output_compressed_accounts: Vec<CompressedAccount>,
+    // index of Merkle tree account in remaining accounts
+    pub output_state_merkle_tree_account_indices: Vec<u8>,
```

Comment:
for these to be fully useful in the mock-rpc (i.e. client/mock-rpc doesn't have to fetch additional data) we'd also have to emit the `output_nullifier_queue_indices`
and put them into the pubkey_array... but onchain doesn't have access to that data AFAIK since its only needed for spending accounts. instead of finding a workaround for this we should prolly make the PublicTransactionEvent smaller again, not emit the output_queues, and not emit any of the redundant info at all either and just implement a few additional steps in the test-rpc.  cc @ananas-block